llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -32387,6 +32390,10 @@ SDValue X86TargetLowering::LowerOperation(SDValue Op, SelectionDAG &DAG) const {
   case ISD::ADDRSPACECAST:      return LowerADDRSPACECAST(Op, DAG);
   case X86ISD::CVTPS2PH:        return LowerCVTPS2PH(Op, DAG);
   case ISD::PREFETCH:           return LowerPREFETCH(Op, Subtarget, DAG);
+  case ISD::CLEAR_CACHE: {
+    // X86 instruction cache is coherent with its data cache so this is a no-op.
+    return Op->getOperand(0);
```

Comment:
Just delete the operation should probably be the default expansion. Probably should move this comment to the setOperationAction, and move this implementation to the default expand action